src/plugins/data/public/antlr/dql/code_completion.test.ts

```diff
@@ -28,82 +42,234 @@ const getSuggestionsAtPos = async (query: string, endPos: number) => {
     language: '', // not relevant
     selectionEnd: 0, // not relevant
     selectionStart: 0, // not relevant
-    services: { appName: 'discover' },
+    services: {
+      appName: 'discover',
+      data: ({
+        autocomplete: { getValueSuggestions: mockValueSuggestions },
+      } as unknown) as DataPublicPluginStart,
+    } as IDataPluginServices,
   });
 };
 
-const getSuggestionAtEnd = async (query: string) => {
+const getSuggestionsAtEnd = async (query: string) => {
   return await getSuggestionsAtPos(query, query.length + 1);
 };
 
+const testAroundClosing = (
```

Comment:
This function contains 2-4 tests that can be triggered around a certain query. These were common patterns found in queries that needed to be tested, basically the behavior around closed/unclosed quotes and parenthesis